electrum/gui/qt/history_list.py

```diff
@@ -247,8 +247,15 @@ def update_fiat(self, row, idx):
         tx_item.update(fiat_fields)
         self.dataChanged.emit(idx, idx, [Qt.DisplayRole, Qt.ForegroundRole])
 
-    def update_item(self, *args):
-        self.refresh('update_item')
+    def update_tx_mined_status(self, tx_hash: str, tx_mined_info: TxMinedInfo):
+        self.tx_status_cache[tx_hash] = self.parent.wallet.get_tx_status(tx_hash, tx_mined_info)
+        try:
+            row = self.transactions.pos_from_key(tx_hash)
```

Comment:
note: we are not using [QStandardItemModel.findItems()](https://doc.qt.io/qt-5/qstandarditemmodel.html#findItems) as it is not defined for QAbstractItemModel and also as that takes linear time in the number of elements in the list